File: IT/L'Indirizzario/C++/Indirizzario C++ v2/MainFrame.cpp
/* file MainFrame.cpp */

#include <QApplication>
#include <QMenuBar>
#include <QVBoxLayout>
#include "MainFrame.h"
#include "Contact.h"
#include "NContact.h"

namespace ind = programmiamo::ind;

ind::MainFrame::MainFrame(QWidget *parent) : QMainWindow(parent) 
{
	/* crea il Notebook */

	tabs = new QTabWidget(this);

	updateUI();

	QMenu *fileMenu = menuBar()->addMenu("&File");

	QAction *saveAction = new QAction("Nuovo", this);
	QAction *removeAction = new QAction("Cancella", this);
	QAction *exitAction = new QAction("Esci", this);

	fileMenu->addAction(saveAction);
	fileMenu->addAction(removeAction);
	fileMenu->addAction(exitAction);

	/* associa a ogni voce del menu il relativo gestore di evento */

	connect(saveAction, SIGNAL(triggered()), this, SLOT(save()));
	connect(removeAction, SIGNAL(triggered()), this, SLOT(remove()));
	connect(exitAction, SIGNAL(triggered()), this, SLOT(exit()));

	QWidget *panel = new QWidget(this);
	this->setCentralWidget(panel);

	QVBoxLayout *box = new QVBoxLayout();
	box->setMargin(0);
	box->addWidget(tabs);
	panel->setLayout(box);

	this->resize(640, 480);
	this->show();
}

/* gestisce il click sul menu "Nuovo", e apre una nuova finestra per inserire un nuovo contatto */

void ind::MainFrame::save()
{
	NContact* window = new NContact(this);
	window->exec();
}

/* gestisce il click sul menu "Cancella": per ora non fa nulla */ 

void ind::MainFrame::remove() {}

/* gestisce il click sul menu "Esci", chiudendo l'applicazione */ 

void ind::MainFrame::exit()
{
	QApplication::quit();
}

/* crea le linguette dei contatti nel Notebook */

void ind::MainFrame::updateUI()
{
	for (int counter = 1; counter < 4; counter++) {
		QString pageName = "Pagina " + QString::number(counter);
		Contact *contact = new Contact(tabs, "Nome", "Cognome", "Indirizzo", "Telefono", "E-mail");
		tabs->addTab(contact, pageName);
	}
}

File: IT/2/L'Indirizzario/Python/Indirizzario Python v2/MainFrame.py
# file MainFrame.py

try:
	from tkinter import Frame, Menu
except ImportError:
	from Tkinter import Frame, Menu

from Pmw import NoteBook
from Contact import * 
from NContact import *

class MainFrame(Frame):
	def __init__(self):
		Frame.__init__(self, height=480, width=640, master=None)

		# crea il Notebook

		self.tabbedPane = NoteBook(self)
		self.updateUI()

		# crea un nuovo menu

		menubar = Menu(self)
		fileMenu = Menu(menubar)
		menubar.add_cascade(label="file", menu=fileMenu)

		# aggiunge tre sottomenu al menu "file", associando a ciascuno il relativo gestore degli eventi

		fileMenu.add_command(label="Nuovo", command=self.save)
		fileMenu.add_command(label="Cancella", command=self.remove)
		fileMenu.add_command(label="Esci", command=self.exit)

		# imposta la barra dei menu

		self.master.config(menu=menubar)

		self.tabbedPane.pack(fill='both', expand=True)

		self.pack_propagate(False)
		self.pack()
		self.master.wm_title("Indirizzario")

	# il metodo save() gestisce il click sul menu "Nuovo"

	def save(self):
		NContact(self)

	# il metodo remove() gestisce il click sul menu "Cancella"

	def remove(self):
		pass

	# il metodo exit() gestisce il click sul menu "Esci"

	def exit(self): 
		self.master.destroy()

	# crea le linguette dei contatti (fittizi) nel Notebook

	def updateUI(self):
		for counter in range(1,4):
			pageName = "Pagina {}".format(counter)
			tab = self.tabbedPane.add(pageName)
			Contact(parent=tab, name="Nome", surname="Cognome", addr="Indirizzo", pnum="Telefono", email="E-mail")

